EDF.py

```diff
@@ -8,32 +41,44 @@
 import re
 import warnings
 
+
 def padtrim(buf, num):
     num -= len(buf)
-    if num>=0:
-        # pad the input to the specified length
-        return str(buf) + ' ' * num
-    else:
-        # trim the input to the specified length
-        return buf[0:num]
-
-####################################################################################################
-# the EDF header is represented as a tuple of (meas_info, chan_info)
-# meas_info should have ['record_length', 'magic', 'hour', 'subject_id', 'recording_id', 'n_records', 'month', 'subtype', 'second', 'nchan', 'data_size', 'data_offset', 'lowpass', 'year', 'highpass', 'day', 'minute']
-# chan_info should have ['physical_min', 'transducers', 'physical_max', 'digital_max', 'ch_names', 'n_samps', 'units', 'digital_min']
-####################################################################################################
+    return (str(buf) + ' ' * num) if (num >= 0) else buf[0:num]
```

Comment:
This syntax is globally accepted and in my system, it has not break in both py2.7 and py3.x yet. You may test it individually and let me know if it does.

Reply:
This is one of the downsides to runtime typing. The syntax is valid, but you might still get an error depending on the types passed.

Reply:
I just checked the code and errors are unlikely. See,

`padtrim()` is a helper function used to standardize header content to pass into `writebyte()`. In Python 2.x, strings and bytes are synonymous unlike in python 3.x. If you check the logic of `writebyte()`, it has error-handling support which will deal with both bytes and string formats i.e. output will be a bytes value. If some other format is returned (such as float or boolean), which is very unlikely logically, then we have to worry. I am not sure how I should write a unit test for least likely scenario

Reply:
There is an advantageous to having consistent types internally. For example, this could be fixed by having the trim case also convert to `str`.  Alternatively, if doing the preferred way of dealing with `str` in Python 3, we assume that everything internally is `str` and only the actual I/O operations are dealing with `bytes` and those are doing the conversion appropriately (then we don't need the conversion to `str` in the pad case). `writebyte` handles the output bit of that, and I think (but haven't checked) that all of the `file.read` are explicitly converted to `str` (or `int`).